### ISSUE TITLE
irc_RPL_MOTD in IRCClient tries to append lines to attribute `motd'. (motd should be a list)

### ISSUE THREAD

**Issue by twisted-trac** [twisted/twisted#3676]
|<img alt="Tyreel's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">| Tyreel reported|
|-|-|
|Trac ID|trac#3676|
|Type|defect|
|Created|2009-02-27 21:24:18Z|
|Branch|https://github.com/twisted/twisted/tree/missing-motdstart-3676|

Small bug, irc_RPL_MOTD method in IRCClient tries to append strings to IRCClients attribute `motd'(which is a string, created at line 524 in irc.py to be a string).

Log:
```
2009/02/27 23:04 +0200 [IRC,client] Unhandled Error
	Traceback (most recent call last):
	  File "/usr/lib/python2.5/site-packages/twisted/internet/tcp.py", line 362, in doRead
	    return self.protocol.dataReceived(data)
	  File "/usr/lib/python2.5/site-packages/twisted/words/protocols/irc.py", line 1478, in dataReceived
	    basic.LineReceiver.dataReceived(self, data.replace('\r', ''))
	  File "/usr/lib/python2.5/site-packages/twisted/protocols/basic.py", line 232, in dataReceived
	    why = self.lineReceived(line)
	  File "/usr/lib/python2.5/site-packages/twisted/words/protocols/irc.py", line 1486, in lineReceived
	    self.handleCommand(command, prefix, params)
	--- <exception caught here> ---
	  File "/usr/lib/python2.5/site-packages/twisted/words/protocols/irc.py", line 1498, in handleCommand
	    method(prefix, params)
	  File "/usr/lib/python2.5/site-packages/twisted/words/protocols/irc.py", line 1112, in irc_RPL_MOTD
	    self.motd.append(params[-1])
	exceptions.AttributeError: 'str' object has no attribute 'append'
```

<details><summary>Searchable metadata</summary>

```
trac-id__3676 3676
type__defect defect
reporter__Tyreel Tyreel
priority__normal normal
milestone__ 
branch__branches_missing_motdstart_3676 branches/missing-motdstart-3676
branch_author__exarkun exarkun
status__closed closed
resolution__fixed fixed
component__words words
keywords__irc irc
time__1235769858000000 1235769858000000
changetime__1236117736000000 1236117736000000
version__None None
owner__ 

```
</details>

**Comment by twisted-trac** [comment]
|<img alt="Tyreel's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Tyreel set owner to @exarkun|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_2"></a>|@exarkun commented|
|-|-|

fixing description markup

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_3"></a>|@exarkun commented|
|-|-|

(In [26319]) Branching to 'missing-motdstart-3676'

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_4"></a>|@exarkun commented|
|-|-|

(In [26320]) handle RPL_MOTD without prior RPL_MOTDSTART

refs [#3676](https://github.com/twisted/twisted/issues/3676)

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)|@exarkun removed owner|
|-|-|

**Comment by twisted-trac** [comment]
|[<img alt="therve's avatar" src="https://avatars.githubusercontent.com/u/159137?s=50" width="50" height="50">](https://github.com/therve)<a name="note_6"></a>|@therve set owner to @exarkun|
|-|-|

Cool, please merge.

**Comment by twisted-trac** [comment]
|[<img alt="exarkun's avatar" src="https://avatars.githubusercontent.com/u/254565?s=50" width="50" height="50">](https://github.com/exarkun)<a name="note_7"></a>|@exarkun set status to `closed`|
|-|-|

(In [26334]) Merge missing-motdstart-3676

Author: exarkun
Reviewer: therve
Fixes: [#3676](https://github.com/twisted/twisted/issues/3676)

Handle the RPL MOTD message when it arrives before the RPL MOTD START message
Previously `IRCClient` would raise an unhandled exception in this case.

**Comment by twisted-trac** [comment]
|<img alt="Automation's avatar" src="https://avatars.githubusercontent.com/u/0?s=50" width="50" height="50">|Automation removed owner|
|-|-|
